HeterogeneousCore/MPICore/plugins/api.cc

```diff
@@ -175,3 +182,14 @@ void MPIChannel::receiveSerializedProduct_(int instance, TClass const* type, voi
   MPI_Mrecv(buffer.Buffer(), size, MPI_BYTE, &message, &status);
   const_cast<TClass*>(type)->ReadBuffer(buffer, product);
 }
+
+void MPIChannel::receiveSimpleProduct_(int instance, edm::ObjectWithDict& product) {
+  int tag = EDM_MPI_SendTrivialProduct | instance * EDM_MPI_MessageTagWidth_;
+  MPI_Message message;
+  MPI_Status status;
+  MPI_Mprobe(dest_, tag, comm_, &message, &status);
+  int size;
+  MPI_Get_count(&status, MPI_BYTE, &size);
+  assert((int)product.typeOf().size() == size);
```

Comment:
Can you change this to use `static_cast` instead of the C-style cast ?
```suggestion
  assert(static_cast<int>(product.typeOf().size()) == size);
```